test/parse.test.ts

```diff
@@ -52,6 +52,52 @@ t.test('Error reporting', async (t) => {
   })
 })
 
+t.test('Selector validation', async (t) => {
+  t.test('passes with valid selectors', async (t) => {
+    const queries = [
+      'diff::changedAny({}, {}, foo)',
+      'diff::changedAny({}, {}, foo.bar.baz)',
+      'diff::changedAny({}, {}, foo[])',
+      'diff::changedAny({}, {}, foo.bar[baz == 1])',
+      'diff::changedAny({}, {}, anywhere(foo))',
+    ]
+
+    queries.forEach((query) => {
+      t.doesNotThrow(() => parse(query))
+    })
+  })
+
+  t.test('fails with invalid selectors', async (t) => {
+    const queries = [
+      'diff::changedAny({}, {}, "foo")',
+      'diff::changedAny({}, {}, 1 + 2)',
+      'diff::changedAny({}, {}, 5)',
+      'diff::changedAny({}, {}, (foo, bar))', // BUG: fails because we don't support tuples yet
+      'diff::changedAny({}, {}, (foo))', // BUG: fails because we don't support groups yet
+    ]
+
+    queries.forEach((query) => {
+      try {
+        parse(query)
+      } catch (error: any) {
+        t.same(error.message, 'Invalid selector syntax')
+      }
+    })
```

Comment:
We prefer native for-of loops for these cases:

```suggestion
    for (const query of queries) {
      try {
        parse(query)
      } catch (error: any) {
        t.same(error.message, 'Invalid selector syntax')
      }
    }
```

Reply:
We can also use [`t.throws`](https://node-tap.org/docs/api/asserts/#tthrowsfn-expectederror-message-extra) here. Something like `t.throws(() => parse(query), 'Invalid selector syntax')`

Reply:
> We prefer native for-of loops for these cases:

Is there a reason why, or is it just our convention?

Reply:
(1) It also supports other types of iterations (e.g. async iteration) so it's more consistent around the code base. (2) It makes it easier to add other control flow inside (e.g. `await` or `return`) if needed.